lib/url.c

```diff
@@ -2834,12 +2834,22 @@ CURLcode Curl_setopt(struct Curl_easy *data, CURLoption option,
     data->set.tcp_keepintvl = va_arg(param, long);
     break;
   case CURLOPT_TCP_FASTOPEN:
-#if defined(CONNECT_DATA_IDEMPOTENT) || defined(MSG_FASTOPEN)
-    data->set.tcp_fastopen = (0 != va_arg(param, long))?TRUE:FALSE;
-#else
-    result = CURLE_NOT_BUILT_IN;
+  {
+    bool supports_fastopen = false;
+#if defined(CONNECT_DATA_IDEMPOTENT)
+#  pragma GCC diagnostic push
+#  pragma GCC diagnostic ignored "-Wpartial-availability"
+    supports_fastopen = (connectx != NULL);
```

Comment:
Do you think perhaps returning `CURLE_FUNCTION_NOT_FOUND` would be better choice? This would require change of specification as well.